gemgis/vector.py

```diff
@@ -19,100 +20,522 @@
 
 """
 
-import geopandas as gpd
-import pandas as pd
-import numpy as np
-import rasterio
-from typing import Union, List, Tuple, Optional, Any
+import pyproj
 import shapely
+import numpy as np
+import pandas as pd
+import geopandas as gpd
 from shapely import geometry
-from scipy.interpolate import griddata, Rbf
 from gemgis.raster import sample
 from gemgis.utils import set_extent
+from scipy.interpolate import griddata, Rbf
+from typing import Union, List, Tuple, Optional, Any
 __all__ = [geometry]
 
+try:
+    import rasterio
+except ModuleNotFoundError:
+    raise ModuleNotFoundError('No valid rasterio installation found')
+
 pd.set_option('display.float_format', lambda x: '%.2f' % x)
 
-# v0.2.0 - Refactoring API
-# def extract_xy_linestrings
-# def extract_xy_points
 
-# Function tested
+def extract_xy_linestrings(gdf: gpd.geodataframe.GeoDataFrame,
+                           reset_index: bool = True,
+                           drop_id: bool = True,
+                           drop_index: bool = True,
+                           drop_points: bool = True,
+                           overwrite_xy: bool = False,
+                           target_crs: str = None,
+                           bbox: List[Union[int, float]] = None) -> gpd.geodataframe.GeoDataFrame:
```

Comment:
```suggestion
                           bbox: Optional[Sequence[float]] = None) -> gpd.geodataframe.GeoDataFrame:
```

Reply:
according to [pep 484](https://www.python.org/dev/peps/pep-0484/#the-numeric-tower),

> Rather than requiring that users write import numbers and then use numbers.Float etc., this PEP proposes a straightforward shortcut that is almost as effective: when an argument is annotated as having type float, an argument of type int is acceptable; similar, for an argument annotated as having type complex, arguments of type float or int are acceptable. This does not handle classes implementing the corresponding ABCs or the fractions.Fraction class, but we believe those use cases are exceedingly rare.